providers/azureblob/src/main/java/org/jclouds/azure/storage/filters/SharedKeyLiteAuthentication.java

```diff
@@ -81,27 +88,82 @@
    @Inject
    public SharedKeyLiteAuthentication(SignatureWire signatureWire,
          @org.jclouds.location.Provider Supplier<Credentials> creds, @TimeStamp Provider<String> timeStampProvider,
-         Crypto crypto, HttpUtils utils) {
+         Crypto crypto, HttpUtils utils, @Named("sasAuth") boolean sasAuthentication) {
       this.crypto = crypto;
       this.utils = utils;
       this.signatureWire = signatureWire;
+      this.storageUrl = URI.create("https://" + creds.get().identity + ".blob.core.windows.net/");
       this.creds = creds;
+      this.identity = creds.get().identity;
+      this.credential = creds.get().credential;
       this.timeStampProvider = timeStampProvider;
+      this.isSAS = sasAuthentication;
    }
-
+   
+   /** 
+   * this is an updated filter method, which decides whether the SAS or SharedKeyLite 
+   * is used and applies the right filtering.  
+   * 
+   */
    public HttpRequest filter(HttpRequest request) throws HttpException {
+      request = this.isSAS ? filterSAS(request, this.credential) : filterKey(request);
+      return request;
+   }
+   
+   /** 
+   * this filter method is applied only for the cases with SAS Authentication. 
+   * 
+   */
+   public HttpRequest filterSAS(HttpRequest request, String credential) throws HttpException {
+      String containerName = null;
+      String blobName = null;
+      URI requestUri = request.getEndpoint();
+	  try { 
+         String[] parametersArray = cutUri(requestUri); 
+         containerName = parametersArray[1]; 
+         UriBuilder endpoint = Uris.uriBuilder(storageUrl).appendPath(containerName);
+         if (parametersArray.length == 3) {
+            endpoint.appendPath(parametersArray[2]).query(credential);
+         } else {
+            endpoint.query("restype=container&" + credential);
+         }
+         return removeAuthorizationHeader(
+            replaceDateHeader(request.toBuilder()
+               .endpoint(endpoint.build())
+               .build()));
+      } catch (NullPointerException e){
+        utils.logRequest(signatureLog, request, "<< a NullPointerException is catched");
+      } catch (IllegalArgumentException ex) {
+        utils.logRequest(signatureLog, request, "<< an IllegalArgumentException is catched ");
+      } 
```

Comment:
If we can't apply the required signature it looks better to fail here and propagate the exception instead of just returning the request to let the request flow continue, sending a request we already know is invalid.

Reply:
Done